src/lexer/mod.rs

```diff
@@ -68,29 +68,45 @@ pub enum TkType {
     Comment,
 }
 
-#[derive(Clone, Debug, PartialEq)]
+#[derive(Clone, Debug)]
 pub struct Location {
     file_name: String,
     line: u32,
     column: u32,
+    pub start: u32,
+    pub end: u32,
 }
 
 impl Location {
     pub fn none() -> Location {
         Location::from(0, 0)
     }
     pub fn from(line: u32, column: u32) -> Location {
-        Location::new("", line, column)
+        Location::new("", line, column, 0, 0)
     }
-    pub fn new<T: ToString>(file_name: T, line: u32, column: u32) -> Location {
+    pub fn new<T: ToString>(
+        file_name: T,
+        line: u32,
+        column: u32,
+        start: u32,
+        end: u32,
+    ) -> Location {
         Location {
             file_name: file_name.to_string(),
             line,
             column,
+            start,
+            end,
         }
     }
 }
 
+impl PartialEq for Location {
+    fn eq(&self, other: &Self) -> bool {
+        self.file_name == other.file_name && self.line == other.line && self.column == other.column
+    }
+}
+
```

Comment:
To prevent the not needed change in testing, locations are only equal to each other when `line`, `column`, and `file_name` are the same.